node/src/client/mod.rs

```diff
@@ -180,6 +229,128 @@ impl<N: Network, C: ConsensusStorage<N>> Client<N, C> {
+                                }
+                                // If error occurs after the first 10 blocks of the epoch, log it as a warning, otherwise ignore.
+                                Err(error) => {
+                                    if _node.ledger.latest_height() % N::NUM_BLOCKS_PER_EPOCH > 10 {
+                                        warn!("Failed to verify the solution - {error}")
+                                    }
+                                }
+                            }
+                        } else {
+                            warn!("Failed to retrieve the latest epoch hash.");
+                        }
+                        // Decrement the verification counter.
+                        _node.solution_verification_counter.fetch_sub(1, Relaxed);
+                    });
+                    // If we are already at capacity, don't verify more solutions.
+                    if previous_counter + 1 >= SOLUTION_MAX_CONCURRENCY {
+                        break;
+                    }
+                }
+            }
+        }));
+    }
+
+    /// Initializes transaction verification.
+    fn initialize_transaction_verification(&self) {
+        // Start the transaction verification loop.
+        let node = self.clone();
+        self.handles.lock().push(tokio::spawn(async move {
+            loop {
+                // If the Ctrl-C handler registered the signal, stop the node.
+                if node.shutdown.load(Acquire) {
+                    info!("Shutting down transaction verification");
+                    break;
+                }
+
+                // Determine if the queue contains txs to verify.
+                let queue_is_empty = node.transaction_queue.lock().is_empty();
+                // Determine if our verification counter has space to verify new txs.
+                let counter_is_full = node.tx_verification_counter.load(Acquire) >= TRANSACTION_MAX_CONCURRENCY;
+
+                // Sleep to allow the queue to be filled or transactions to be validated.
+                if queue_is_empty || counter_is_full {
+                    sleep(Duration::from_millis(50)).await;
+                    continue;
+                }
+
+                // Try to verify transactions.
+                let mut tx_queue = node.transaction_queue.lock();
+                while let Some((_, (peer_ip, serialized, transaction))) = tx_queue.pop_lru() {
+                    // Increment the verification counter.
+                    let previous_counter = node.tx_verification_counter.fetch_add(1, Relaxed);
+                    let _node = node.clone();
+                    // For each transaction, spawn a task to verify it.
+                    tokio::spawn(async move {
+                        // Check the transaction.
+                        if _node.ledger.check_transaction_basic(&transaction, None, &mut rand::thread_rng()).is_ok() {
```

Comment:
shouldn't this be done in the context of a blocking task?

Reply:
Makes sense to me: [799f157](https://github.com/AleoNet/snarkOS/pull/3358/commits/799f157ceaccd768ca26ddab340fae3b3f09bd26)